pkg/container-collection/container-collection.go

```diff
@@ -127,8 +134,20 @@ func (cc *ContainerCollection) RemoveContainer(id string) {
 		return
 	}
 
+	container := v.(*Container)
+
 	if cc.pubsub != nil {
-		cc.pubsub.Publish(EventTypeRemoveContainer, v.(*Container))
+		cc.pubsub.Publish(EventTypeRemoveContainer, container)
+	}
+
+	// Save the container in the cache as enrichers might need the container some time after it
+	// has been removed.
+	if cc.cachedContainers != nil {
+		cc.cachedContainers.Store(id, v)
+		time.AfterFunc(cc.cacheDelay, func() {
+			cc.cachedContainers.Delete(id)
+			unix.Close(container.mntNsFd)
```

Comment:
This is fine for gadgettracermanager and local-gadget because they use `WithTracerCollection` but in general mntNsFd is not initialized if `WithTracerCollection` is not used.

Reply:
It's only executed if `cc.cachedContainers != nil` and there is a condition on L145 checking it.